core/lib/eth_client/src/lib.rs

```diff
@@ -179,7 +183,7 @@ static_assertions::assert_obj_safe!(EthInterface);
 /// 2. Consider adding the "unbound" version to the `EthInterface` trait and create a default method
 /// implementation that invokes `contract` / `contract_addr` / `sender_account` methods.
 #[async_trait]
-pub trait BoundEthInterface: EthInterface {
+pub trait BoundEthInterface: AsRef<dyn EthInterface> + 'static + Sync + Send + fmt::Debug {
```

Comment:
Not sure whether this is controversial; IMO, `EthInterface` and `BoundEthInterface` encapsulate "adjacent" functionality, so it doesn't really make sense to require one inherit from the other.